runtime/vm/BytecodeInterpreter.hpp

```diff
@@ -5114,16 +5115,30 @@ class INTERPRETER_CLASS
 			}
 		}
 
+		bp = buildSpecialStackFrame(REGISTER_ARGS, J9SF_FRAME_TYPE_JNI_NATIVE_METHOD, jitStackFrameFlags(REGISTER_ARGS, 0), true);
+		*--_sp = (UDATA)_sendMethod;
+		_arg0EA = bp + 8;
 		updateVMStruct(REGISTER_ARGS);
 		VM_VMAccess::inlineExitVMToJNI(_currentThread);
+		VM_VMHelpers::beforeJNICall(_currentThread);
 #if FFI_NATIVE_RAW_API
 		ffi_ptrarray_to_raw(cif, values, values_raw);
 		ffi_raw_call(cif, FFI_FN(function), returnStorage, values_raw);
 #else /* FFI_NATIVE_RAW_API */
 		ffi_call(cif, FFI_FN(function), returnStorage, values);
 #endif /* FFI_NATIVE_RAW_API */
+		VM_VMHelpers::afterJNICall(_currentThread);
 		VM_VMAccess::inlineEnterVMFromJNI(_currentThread);
 		VMStructHasBeenUpdated(REGISTER_ARGS);
+		if (VM_VMHelpers::exceptionPending(_currentThread)) {
+			rc = GOTO_THROW_CURRENT_EXCEPTION;
+		}
+		{
+			bp = _arg0EA - 8;
```

Comment:
What is this magic 8?

Reply:
This is the slot count of arguments for this INL method `invokeNative(long returnStructMemAddr, long functionAddr, long calloutThunk, long[] argValues)` (see. the INL method `inlAttachmentLoadAgentLibraryImpl` as follows:)
```
	/* openj9.internal.tools.attach.target.Attachment: private native int loadAgentLibraryImpl(ClassLoader loader, String agentLibrary, String options, boolean decorate); */ <---- argument slots are 5 for this INL method
	VMINLINE VM_BytecodeAction
	inlAttachmentLoadAgentLibraryImpl(REGISTER_ARGS_LIST)
...
		UDATA *bp = buildSpecialStackFrame(REGISTER_ARGS, J9SF_FRAME_TYPE_JNI_NATIVE_METHOD, jitStackFrameFlags(REGISTER_ARGS, J9_STACK_FLAGS_USE_SPECIFIED_CLASS_LOADER), true);
		*--_sp = (UDATA)_sendMethod;
		_arg0EA = bp + 5; <----
...
		bp = _arg0EA - 5; <----
``` 
So I replaced `8` with `argSlots` (set to `8` in code) to avoid confusion.
```
		UDATA argSlots = 8;
```